test/github.js

```diff
@@ -31,157 +31,158 @@ const valid = {
   //
   // NOTE these do not accept auth at all
   'foo/bar': { ...defaults, default: 'shortcut' },
-  'foo/bar#branch': { ...defaults, default: 'shortcut', committish: 'branch' },
+  'foo/bar#branch:^1.0.0': { ...defaults, default: 'shortcut', committish: 'branch:^1.0.0' },
```

Comment:
Since the original bug reported in #148 was only present in some urls, I replaced all `committish` assertions with a semver looking branch name to be sure I caught any other code paths that might have had a similar problem. I don't think anything is lost by using this in all cases.

Reply:
Are there other special characters in branches we want to enforce regression testing on? `/`?

Reply:
Good call on this. `@` is valid in branches and we were breaking on those too.